Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -564,69 +566,66 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
 
-    open override func startTunnel(options: [String: NSObject]?, completionHandler: @escaping (Error?) -> Void) {
-        Task { @MainActor in
-            providerEvents.fire(.tunnelStartAttempt(.begin))
-            prepareToConnect(using: tunnelProviderProtocol)
-
-            connectionStatus = .connecting
+    @MainActor
+    open override func startTunnel(options: [String: NSObject]? = nil) async throws {
 
-            let startupOptions = StartupOptions(options: options ?? [:])
-            os_log("Starting tunnel with options: %{public}s", log: .networkProtection, startupOptions.description)
+        let startupOptions = StartupOptions(options: options ?? [:])
+        os_log("Starting tunnel with options: %{public}s", log: .networkProtection, startupOptions.description)
 
-            resetIssueStateOnTunnelStart(startupOptions)
+        do {
+            try load(options: startupOptions)
+            try loadVendorOptions(from: tunnelProviderProtocol)
+        } catch {
+            if startupOptions.startupMethod == .automaticOnDemand {
+                // If the VPN was started by on-demand without the basic prerequisites for
+                // it to work we skip firing pixels.  This should only be possible if the
+                // manual start attempt that preceded failed, or if the subscription has
+                // expired.  In either case it should be enough to record the manual failures
+                // for these prerequisited to avoid flooding our metrics.
+                try? await Task.sleep(interval: .seconds(15))
+            } else {
+                // If the VPN was started manually without the basic prerequisited we always
+                // want to know as this should not be possible.
+                providerEvents.fire(.tunnelStartAttempt(.begin))
+                providerEvents.fire(.tunnelStartAttempt(.failure(error)))
+            }
 
-            let internalCompletionHandler = { [weak self, providerEvents] (error: Error?) in
-                guard let error else {
-                    completionHandler(nil)
-                    providerEvents.fire(.tunnelStartAttempt(.success))
-                    return
-                }
+            os_log("🔴 Stopping VPN due to no auth token: %{public}s", log: .networkProtection)
+            await attemptShutdown()
 
-                let handler = {
-                    let errorDescription = (error as? LocalizedError)?.localizedDescription ?? String(describing: error)
+            throw error
+        }
 
-                    os_log("Tunnel startup error: %{public}@", type: .error, errorDescription)
-                    self?.controllerErrorStore.lastErrorMessage = errorDescription
-                    self?.connectionStatus = .disconnected
-                    self?.knownFailureStore.lastKnownFailure = KnownFailure(error)
+        do {
+            providerEvents.fire(.tunnelStartAttempt(.begin))
+            prepareToConnect(using: tunnelProviderProtocol)
```

Comment:
We should invoke this as soon as possible, as `PixelKit` is expected to be set up within an overriden `prepareToConnect` implementation.

Reply:
Well, that's a very good observation.  Moved to the top.